json_tokener.c

```diff
@@ -543,6 +543,12 @@ struct fjson_object* fjson_tokener_parse_ex(struct fjson_tokener *tok,
 		    printbuf_memappend_fast(tok->pb, (char*)utf8_replacement_char, 3);
                   }
                   got_hi_surrogate = 0;
+                  /* clang static analyzer thins that got_hi_surrogate is never read,
+		   * however, it is read on each iteration. So I assume clang has a false
+		   * positive. We use the otherwise nonsense statement below to make it
+		   * happy.
+		   */
```

Comment:
2016-04-08 18:27 GMT+02:00 Michael Biebl <notifications@github.com>:

> In json_tokener.c
> <https://github.com/rsyslog/libfastjson/pull/44#discussion_r59051543>:
>
> > @@ -543,6 +543,12 @@ struct fjson_object* fjson_tokener_parse_ex(struct fjson_tokener *tok,
> >  		    printbuf_memappend_fast(tok->pb, (char*)utf8_replacement_char, 3);
> >                    }
> >                    got_hi_surrogate = 0;
> > +                  /* clang static analyzer thins that got_hi_surrogate is never read,
> > +		   * however, it is read on each iteration. So I assume clang has a false
> > +		   * positive. We use the otherwise nonsense statement below to make it
> > +		   * happy.
> > +		   */
>
> What's the error message from clang?
>

Have a look here:
   https://travis-ci.org/rgerhards/libfastjson/jobs/121737854

Isn't there a better way to tell clang to shut up?
>

Unfortunately not, clang has not option to suppress warnings. See also:

https://developer.gnome.org/programming-guidelines/stable/tooling.html.en

I already banged my head on that warning a couple of days ago, and don't
see any chance that clang is right. It also doesn't give any path info in
interactive viewer, just the same message.